### PR TITLE
Change to GitHub actions

### PR DESCRIPTION
**Pull request by wxtim**
## Issue

Tunebook hasn't been building on Travis

## Work undertaken

- [x] Moved to using Github actions.

## How to test

Trial Deployed to https://wxtim.github.io/bristolfolksoc.github.io/tunes.html

## Other

@MorrisNerd2  I'm not sure whether you might want to look at https://github.com/wxtim/bristolfolksoc.github.io/runs/6747568422?check_suite_focus=true  and look at > create tunebook. I can't say for sure whether it's complaining about files because the files are wrong or because the program is - although when I use abcm2ps it's usually me. Looking quickly at "Vandals of Hammerwich"  (just picked one at random) and [the abc standard](https://abcnotation.com/wiki/abc:standard:v2.1#chords_and_unisons) I think that you might have failed to close some brackets around notes in chords:

```diff
 P: Song A
-"G/A" [F4DA, [E4CA,|"D"A,2 D2 F2 D2|...
+"G/A" [F4DA,] [E4CA,]|"D"A,2 D2 F2 D2|...
 P: Normal A

```

### CODE REVIEW

File: travis/deploy.sh [hunk 1]
```diff
@@ -1,26 +1,16 @@
...
+mv node_modules/ node_modules_full/
+npm install --production
+git config --global user.email "deploy@travis-ci.org"
+git config --global user.name "Deployment Bot"
+rm .gitignore
+mv .gitignore-deploy .gitignore
+git checkout -b master
+git add -A
+git commit -m "Build"
+git remote add origin-pages https://git@github.com/wxtim/bristolfolksoc.github.io.git > /dev/null 2>&1
```
Comment (wxtim) [inline travis/deploy.sh]: Needs change before final merge.

```suggestion
git remote add origin-pages https://git@github.com/bristolfolksoc/bristolfolksoc.github.io.git > /dev/null 2>&1
```

File: travis/deploy.sh [hunk 2]
```diff
@@ -1,26 +1,16 @@
 #!/bin/bash
 # This script pushes the travis build to the master branch if we are on the development branch
-if [ "$TRAVIS_PULL_REQUEST" == "false"  ]; then
+set -eu
```
Comment (wxtim) [inline travis/deploy.sh]: This forces the bash script to fail if something goes wrong!

File: .github/workflows/buildtunebook.yaml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+          mv node_modules/ node_modules_full/
+          npm install --production
+          git config --global user.email "deploy@travis-ci.org"
+          git config --global user.name "Deployment Bot"
+          rm .gitignore
+          mv .gitignore-deploy .gitignore
+          git checkout -b master
+          git add -A
+          git commit -m "Build"
+          git remote add origin-pages https://git@github.com/wxtim/bristolfolksoc.github.io.git > /dev/null 2>&1
```
Comment (wxtim) [inline .github/workflows/buildtunebook.yaml]: Before merge

```suggestion
          git remote add origin-pages https://git@github.com/bristolfolksoc/bristolfolksoc.github.io.git > /dev/null 2>&1
```

File: tunes/Delvin House.abc [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+R: Strathspey
+M: C
+L: 1/16
+K: C
+r:16
+C: Trad
+|: e2 | c4 G3e c4 G3E | Dd3d3c BG3A3B | c3ed3e c3eA3c | G3-AG3-F Gc3 c4 :| 
+e2 |: {^f}g3ec3e cdef g3e | f3dB3d Bcde e3d |1 g3-ec3-e g3ec3G |
+Af3 e3d e3cc3e :|2 eg3 de3 ce3 Ac3| G3-AG3F Gc3 c2 ||
+Z: Mackin, from a cute old book of hundreds of tunes from her Violin teacher's Grandmother.
```
Comment (wxtim) [inline tunes/Delvin House.abc]: a line at the end of a file is good practice @MorrisNerd2

File: .github/workflows/buildtunebook.yaml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+      - name: cache npm
+        uses: actions/cache@v2
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
+
+      - name: pre install
+        run: |
+          sudo apt install abcm2ps
```
Comment (wxtim) [inline .github/workflows/buildtunebook.yaml:33]: @Nick-Pearson In theory using sudo creates a vulnerability to change, however I think these very old scripts are being maintained well - they seem the basis of far too many other ABC programs. I've left the sub-modules for safety's sake, lest we need to go back to the old method (although in theory the VC system should make this academic).

Does this seem reasonable to you?

File: tunes/Delvin House.abc (hunk 4 above)
Comment (Nick-Pearson) [reply tunes/Delvin House.abc]: iirc one of these libraries we use for the book creation is also able to lint abc files which gives the option of checking the files in gh actions as well

File: .github/workflows/buildtunebook.yaml [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+          ./epstopdf/epstopdf -V
+          npm install
+
+      - name: create tunebook
+        run: |
+          node generate.js index
+          node generate.js books
+          npm test;
+
+      - name: deploy tunebook
```
Comment (Nick-Pearson) [inline .github/workflows/buildtunebook.yaml:50]: Will this be run on a pull request?

File: .github/workflows/buildtunebook.yaml (hunk 5 above)
Comment (Nick-Pearson) [reply .github/workflows/buildtunebook.yaml:33]: Yes seems fair enough to me

File: tunes/DingDongMerrilyonHigh.abc [hunk 7]
```diff
@@ -5,7 +5,6 @@ L:1/16
 M:2/4
 K:G
 R: Christmas 
-r: 24
```
Comment (Nick-Pearson) [inline tunes/DingDongMerrilyonHigh.abc]: this is read by generate.js as the number of bars

**Comment by Nick-Pearson** [comment]
The latex tunebook pdfs rely on this very Travis specific dockerfile https://github.com/Strauman/travis-latexbuild
We will need to get a github actions equivalent (perhaps that should be done separately to this pr?)

**Comment by MorrisNerd2Again** [comment]
Last one: Yeah let's do it as a separate one.

File: .github/workflows/buildtunebook.yaml (hunk 6 above)
Comment (wxtim) [reply .github/workflows/buildtunebook.yaml:50]: Yes - it only wants to be on merge, doesn't it!

File: tunes/Delvin House.abc (hunk 4 above)
Comment (wxtim) [reply tunes/Delvin House.abc]: I created a separate issue for an on PR change at https://github.com/bristolfolksoc/bristolfolksoc.github.io/issues/22

File: tunes/DingDongMerrilyonHigh.abc (hunk 7 above)
Comment (wxtim) [reply tunes/DingDongMerrilyonHigh.abc]: This is a rebase error.

**Comment by wxtim** [comment]
I've got some changes to make (this should only run to merge with dev branch I think). I'll do that out of work time.

File: .github/workflows/buildtunebook.yaml (hunk 6 above)
Comment (wxtim) [reply .github/workflows/buildtunebook.yaml:50]: I've changed it to 
```
on:
  push:
    branches:
      - development
```

Which is functionally equivalent to on merge (which doesn't exist) unless the developer has direct push permissions.

**Review by Nick-Pearson — APPROVED** [review]

**Comment by Nick-Pearson** [comment]
Looks good to me
Thanks @wxtim!

**Pull request merged**